src/Unused/CLI/Views/SearchResult.hs

```diff
@@ -51,9 +53,14 @@ likelihoodColor Low = Green
 likelihoodColor Unknown = Black
 likelihoodColor NotCalculated = Magenta
 
-printMatches :: ColumnFormat -> TermResults -> [TermMatch] -> IO ()
-printMatches cf r ms =
-    forM_ ms $ \m -> do
+printMatches :: TermResults -> [TermMatch] -> ReaderT ColumnFormat IO ()
+printMatches r ms = do
+    cf <- ask
+    let printTerm = cfPrintTerm cf
+    let printPath = cfPrintPath cf
+    let printNumber = cfPrintNumber cf
```

Comment:
because these are partially applied functions used within `forM_` below, there's no clean way to go about moving this down.

Ideally, I'd have been able to do:

```haskell
configPrintTerm :: String -> Reader ColumnFormat String
configPrintTerm v = ask >>= return . flip cfPrintTerm v

-- more

    liftReaderT $ forM_ ms $ \m -> do
        setSGR [SetColor Foreground Dull (termColor r)]
        setSGR [SetConsoleIntensity NormalIntensity]
        putStr $ "     " ++ configPrintTerm (tmTerm m)
        setSGR [Reset]
```

But instead, to do that it requires:

```haskell
    liftReaderT $ forM_ ms $ \m -> do
        setSGR [SetColor Foreground Dull (termColor r)]
        setSGR [SetConsoleIntensity NormalIntensity]
        putStr $ "     " ++ runReader (configPrintTerm $ tmTerm m) cf
        setSGR [Reset]
```

This is because `liftReaderT` effectively removes the environment when converting from `IO`, so `runReader` is required to re-provide it.

I'm interested in other ways to solve this, though.